machida/lib/wallaroo/__init__.py

```diff
@@ -252,7 +252,7 @@ def encode(self, data, event_time=0, key=None):
                     encoded_key = key.encode() if key else ''.encode()
                     return struct.pack(
                         '>IqI{}s{}s'.format(len(encoded_key), len(encoded)),
-                        len(encoded) + 8 + len(encoded_key) + 8, # total frame size
+                        len(encoded) + 8 + len(encoded_key) + 4, # total frame size
```

Comment:
Something's still looks wrong here...
I think the second `I` in the format string shouldn't be there?
```
I: 4 bytes
q: 8 bytes
len(encoded_key): encoded_key
len(encoded): encoded
```

Reply:
Nevermind. I'm wrong.

Reply:
I think the original version was correct: there are:
2 x 4 bytes (`I`)
1x 8 bytes (`q`)

Reply:
Hmmm, on reflection, if I am right, then (without the patch) Python is allowing a buffer overrun, right?

Reply:
```2 x 4 bytes (I)
1x 8 bytes (q)```

It ought to be `1 x 4 bytes (I)`: the bytes of first `I` belong to the message header not the body.

If I append a `|` byte to the end of the packet without doing anything to adjust the header length, then I see two different behaviors:

* unpatched: the 1st message received includes the `|` byte plus 3 additional NUL bytes
* patched: the 1st message received is the proper length (4 bytes of app payload), and the 2nd message received has a nonsense length because the `|` is the first frame header length byte.

Reply:
From slack conversation:
- [ ] let's reorder the additions there to match the order of items in the format spring
- [ ] maybe add a comment to remind future us that the first `I`, being the length header, should not count toward the length value.